pkg/apis/openebs.io/v1alpha1/cstor_volume_claim.go

```diff
@@ -0,0 +1,91 @@
+/*
+Copyright 2019 The OpenEBS Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +k8s:openapi-gen=true
+
+// CStorVolumeClaim describes a cstor volume claim resource created as
+// custom resource. CStorVolumeClaim is a request for creating cstor volume
+// related resources like deployment, svc etc.
+type CStorVolumeClaim struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+	// Spec defines a specification of a cstor volume claim required
+	// to provisione cstor volume resources
+	Spec CStorVolumeClaimSpec `json:"spec"`
+
+	// Publish contains info related to attachment of a volume to a node.
+	// i.e. NodeId etc.
+	Publish map[string]string `json:"publish,omitempty"`
```

Comment:
this is supposed to be just string

Reply:
or make it as a struct with `nodeId` being a string in it

Reply:
```yaml
kind: CStorVolumeClaim
metadata:
  name: <name of the volume>
  namespace: <openebs system namespace>
  labels:
  annotations:
    openebs.io/config-class: 
    openebs.io/volume-id:
spec:
  capacity:
  claimRef:
publish:
  nodeId:
status:
  phase: # INIT, BOUND and ERROR can be the supported phases

  # conditions represent current reconciliation
  # activities
  conditions:
  - type:  # RESIZE_IN_PROGRES, etc. TODO Need to think of proper names.
    active:
    message:
    createdAt:
    lastUpdatedAt:
    count:
```

Reply:
If we just need the `NodeID` or `Name` we can just use the string.

Reply:
Done